### ISSUE TITLE
ExecutablePath option ignored

### ISSUE THREAD

**Issue by nc** [puppeteer/puppeteer#6614]
MacOS / BigSur
Node 16 (Head)
puppeteer-core, 5.5.0

```
const puppeteer = require('puppeteer-core');
const browser = await puppeteer.launch({
  product: 'chrome',
  headless: false,
  executablePath: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
});
```

Fails with 

`Error: Failed to launch the browser process! spawn /usr/bin/chromium-browser ENOENT`

As I understand it, it should not be trying to launch chromium at all given the above options and use of puppeteer-core.

**Comment by nc** [comment]
If however, one modifies the BrowserRunner directly to hardcode the executablePath as shown above, all works as expected. So it seems to be an issue with how the option is being passed to Puppeteer, or how it's being handled. 
```
class BrowserRunner {
    constructor(executablePath, processArguments, tempDirectory) {
        this.proc = null;
        this.connection = null;
        this._closed = true;
        this._listeners = [];
        this._executablePath = executablePath;
        this._processArguments = processArguments;
        this._tempDirectory = tempDirectory;
    }
    start(options) {
        console.log('>>>>', this._executablePath);
        this._executablePath = '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';
```
This prints
`>>>> /usr/bin/chromium-browser`
and then proceeds to launch Google Chrome as expected.

**Comment by eugene1g** [comment]
This happens on Apple Silicon. The path is hardcoded at https://github.com/puppeteer/puppeteer/blob/d901696e0d8901bcb23cf676a5e5ac562f821a0d/src/node/Launcher.ts#L106-L109  introduced in https://github.com/puppeteer/puppeteer/commit/354f9424ae4014e0cd276a930d86a6dae8fe427e

Chromium can run natively on Apple Silicon. Puppeteer does not yet link to that binary, but if the developer uses `executablePath` to point to another Chrome instance (that might not be compatible with Puppeteer), they already take on those risks and the library should just accept that path.

**Comment by eugene1g** [comment]
Related https://github.com/puppeteer/puppeteer/issues/6641 https://github.com/puppeteer/puppeteer/pull/6495 https://github.com/puppeteer/puppeteer/issues/6634

**Comment by PrzemekCh** [comment]
> If however, one modifies the BrowserRunner directly to hardcode the executablePath as shown above, all works as expected. So it seems to be an issue with how the option is being passed to Puppeteer, or how it's being handled.
> 
> ```
> class BrowserRunner {
>     constructor(executablePath, processArguments, tempDirectory) {
>         this.proc = null;
>         this.connection = null;
>         this._closed = true;
>         this._listeners = [];
>         this._executablePath = executablePath;
>         this._processArguments = processArguments;
>         this._tempDirectory = tempDirectory;
>     }
>     start(options) {
>         console.log('>>>>', this._executablePath);
>         this._executablePath = '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';
> ```
> 
> This prints
> `>>>> /usr/bin/chromium-browser`
> and then proceeds to launch Google Chrome as expected.

You saved my day :)

**Comment by thamwangjun** [comment]
I think this is a duplicate of #6634, and this issue has been fixed in this merge request #6495.
In the latest puppeteer version `10.4.0`, `executablePath` is working in my tests.

**Comment by behemoth-thainv** [comment]
I'm using `puppeteer-core@^5.2.1`.
Here is solution if you won't update version:

### Get chronium path on your device:
```
which chromium
```
<img width="661" alt="Screenshot 2023-10-03 at 15 45 11" src="https://github.com/puppeteer/puppeteer/assets/60531136/48005d74-66c2-46a2-9c9e-2d0b01cd2720">


## Modify default `executablePath` for `arm64`
```
path: node_modules/puppeteer-core/lib/cjs/puppeteer/node/Launcher.js
change: '/usr/bin/chromium-browser' -> '/opt/homebrew/bin/chromium'
```
<img width="1318" alt="image" src="https://github.com/puppeteer/puppeteer/assets/60531136/6e16a8a9-71e6-4114-9f36-196c5447b285">
